include/parser/riscv/riscv-regex.hpp

```diff
@@ -0,0 +1,125 @@
+/* C++ Assembler Interpreter
+ * Copyright (C) 2016 Chair of Computer Architecture
+ * at Technical University of Munich
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.*/
+
+#ifndef ERAGPSIM_PARSER_RISCV_RISCV_REGEX_HPP
+#define ERAGPSIM_PARSER_RISCV_RISCV_REGEX_HPP
+
+#include <regex>
+#include <string>
+
+#include "parser/riscv/riscv-parser.hpp"
+
+class CompileErrorList;
+#include "parser/independent/positioned-string.hpp"
+
+/**
+ * Wrapper class for the RegExp used for parsing.
```

Comment:
This class looks like it does its own string processing/matching. Why not use regular expressions?